openedx_learning/core/components/api.py

```diff
@@ -98,10 +166,84 @@ def create_component_and_version(
         )
         return (component, component_version)
 
+def get_component(component_pk: int) -> Component:
+    """
+    Get Component by its primary key.
+
+    This is the same as the PublishableEntity's ID primary key.
+    """
+    return Component.with_publishing_relations.get(pk=component_pk)
 
-def get_component_by_pk(component_pk: int) -> Component:
-    return Component.objects.get(pk=component_pk)
+def get_component_by_key(
+    learning_package_id: int,
```

Comment:
Good point. Maybe I should stick with `_pk` for everything?

The reason why it's not `component_id` is because there is no `Component.id` field–Components have a `OneToOneField` relationship to `PublishableEntity` with `primary_key=True`. So the actual database field is `oel_components_component.publishable_entity_id`.

So maybe I should just convert everything to be `_pk` for consistency instead, since `learning_package.pk` will just be an alias to `learning_package.id` anyway. But one of the things that I didn't think through at the time is that if you have a foreign key relation to `Component` from another model, the db field will be `component_id`, and `component_version.component_pk` doesn't work.

Reply:
Personally I don't think it's a big deal either way, but I'd probably lean toward calling it `component_id` for consistency, even though `Component.id` doesn't exist.

Reply:
@ormsbee @bradenmacdonald Consider me a soft vote for `_pk`. In edx-platform, `_id` could indicate all sorts of things: primary db keys, opaque keys (course_id, library_id, usage_id), and other things (block_id, user_id). `_pk` is unambiguously a primary db key.

Reply:
Yeah, I'm going to go with `_pk` everywhere and do that refactoring now.